### PR TITLE
Android permissions and Gradle changes

### PR DESCRIPTION
**Pull request by asafron**
**Android Gradle:**
Remvoed custom clean task as it is not allowed in newer Gradle version.

**Permissions:**
When calling StartSpeech, if RECORD_AUDIO permission was not granted, it will ask the user for permission and will return (will not start recording automatically after user has granted the permission).

### CODE REVIEW

**Comment by wenkesj** [comment]
@asafron this is a good idea, I think we should add a flag to allow this functionality for backwards compatibility. I am making some small changes and merging some forks. This will come in the next release tonight (will merge soon)

**Comment by asafron** [comment]
@wenkesj this is awesome, please let me know if I can help somehow to make the process easier for you.

File: android/src/main/java/com/wenkesj/voice/VoiceModule.java [hunk 1]
```diff
@@ -43,6 +44,26 @@ public String getName() {
 
   @ReactMethod
   public void startSpeech(final String locale, final Callback callback) {
+    if (!isPermissionGranted()) {
```
Comment (wenkesj) [inline android/src/main/java/com/wenkesj/voice/VoiceModule.java]: I like this idea here it's very graceful, but I think still this should be a flag on the new options object passed with `Voice.start`. For instance:
```js
if (!isPermissionGranted() && options.getBool("REQUEST_PERMISSIONS_AUTO")) {
// ...
}
```
.

**Comment by asafron** [comment]
@wenkesj 2 more issues which came up in my mind during fixing all the conflicts and updating the JS and README files:
1. I think we should add a standalone method for requesting user permission.
2. If `REQUEST_PERMISSIONS_AUTO:` true, you think we should start listening immediately (without the need to call `startSpeech` again)? I think the answer is yes, but wanted to confirm it before I implement it (didn't appear in my original PR).

File: VoiceTest/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
 	"dependencies": {
 		"react": "16.0.0-beta.5",
 		"react-native": "0.49.2",
-    "react-native-voice": "0.2.0"
+		"react-native-voice": "file:.."
```
Comment (wenkesj) [inline VoiceTest/package.json]: This might not belong in this PR, but honestly, this is a good idea... In reality, the `VoiceTest` should have the file location from local changes. But for observers and others using the `VoiceTest` with their local changes, I think keeping the npm version on the `master` branch is the way to go.

Comment (asafron) [reply VoiceTest/package.json]: my bad, will deploy a fix soon

**Comment by asafron** [comment]
@wenkesj It was a bit challenging to configure everything for local development, but now it seems ok. Will appriciate it if you could review the changes and let me know if you think something is missing. After merging this PR I'll add a different PR with the 2 issues I've mentioned in my previous comment. Sounds good?

**Comment by wenkesj** [comment]
@asafron LGTM

**Comment by meiqi1992** [comment]
@wenkesj react-antive 0.57.0 android get error ‘network error’，but ios work well？how to resolve?
